src/Temporalio.Extensions.Hosting/TemporalWorkerClientUpdater.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Temporalio.Worker;
+
+namespace Temporalio.Extensions.Hosting
+{
+    /// <summary>
+    /// Notification hub that can be used to push Temporal worker client updates to subscribing Temporal workers.
+    /// </summary>
+    public class TemporalWorkerClientUpdater
+    {
+        private readonly object clientLock = new();
+
+        private event EventHandler<IWorkerClient>? OnClientUpdatedEvent;
+
+        /// <summary>
+        /// Dispatches a notification to all subscribers that a new worker client should be used.
+        /// </summary>
+        /// <param name="client">The new <see cref="IWorkerClient"/> that should be pushed out to all subscribing workers.</param>
+        public void UpdateClient(IWorkerClient client)
+        {
+            OnClientUpdatedEvent?.Invoke(this, client);
+        }
+
+        /// <summary>
+        /// Adds a new subscriber that will be notified when a new worker client should be used.
+        /// </summary>
+        /// <param name="eventHandler">The event handler to add to the event listeners.</param>
+        internal void Subscribe(EventHandler<IWorkerClient> eventHandler)
+        {
+            lock (clientLock)
+            {
+                OnClientUpdatedEvent += eventHandler;
+            }
```

Comment:
Looks like locking technically not required per https://stackoverflow.com/questions/22825117/is-it-thread-safe-to-register-for-a-c-sharp-event, but this is fine